include/xsparse/level_capabilities/co_iteration.hpp

```diff
@@ -155,20 +171,30 @@ namespace xsparse::level_capabilities
         public:
             Coiterate const& m_coiterate;
             std::tuple<Is...> const m_i;
-            PK const m_pkm1;
+            std::tuple<Ps...> const m_pkm1;
             std::tuple<typename Levels::iteration_helper...> m_iterHelpers;
 
         public:
             explicit inline coiteration_helper(Coiterate const& coiterate,
                                                std::tuple<Is...> i,
-                                               PK pkm1) noexcept
+                                               std::tuple<Ps...> pkm1) noexcept
                 : m_coiterate(coiterate)
                 , m_i(std::move(i))
                 , m_pkm1(std::move(pkm1))
-                , m_iterHelpers(std::apply([&](auto&... args)
-                                           { return std::tuple(args.iter_helper(i, pkm1)...); },
-                                           coiterate.m_levelsTuple))
+                // , m_iterHelpers(std::apply([&](auto&... args)
+                // {
+                //     return std::tuple(args.iter_helper(std::get<0>(m_i), m_pkm1)...);
+                // }, coiterate.m_levelsTuple))
+                , m_iterHelpers(unfold_and_apply_helper(coiterate.m_levelsTuple,
+                                                        std::get<0>(m_i),
+                                                        pkm1,
+                                                        std::index_sequence_for<Ps...>{}))
```

Comment:
This is the new change that leverages a templated `unfold_and_apply_helper` function to define the iter_helpers for each `coiterate.m_levelsTuple`.